### PR TITLE
Import sapling global parameters

### PR DESCRIPTION
**Pull request by ebfull**
This will also test the API a bit by invoking the Pedersen hashes.

Part of #3053.

### CODE REVIEW

**Comment by ebfull** [comment]
@zkbot try

**Comment by str4d** [comment]
Passed on Debian 8 and Debian 9, but failed on Ubuntu 16.04:

```
libtool: link: /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/share/../native/bin/ccache g++ -m64 -std=c++11 -Wformat -Wformat-security -Wstack-protector -fstack-protector-all -Werror -fPIE -pipe -O1 -g -fopenmp -fwrapv -fno-strict-aliasing -Wno-builtin-declaration-mismatch -pthread -Wl,-z -Wl,relro -Wl,-z -Wl,now -pie -o zcashd zcashd-bitcoind.o -pthread -pthread  -L/home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/share/../lib libbitcoin_server.a libbitcoin_common.a univalue/.libs/libunivalue.a libbitcoin_util.a libzcash.a snark/libsnark.a ./leveldb/libleveldb.a ./leveldb/libmemenv.a secp256k1/.libs/libsecp256k1.a libbitcoin_zmq.a -L/home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libzmq.a -lrt -lpthread -lstdc++ libbitcoin_wallet.a -lboost_filesystem -lboost_program_options -lboost_thread -lboost_chrono -ldb_cxx-6.2 -lssl -ldl /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libevent_pthreads.a /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libevent.a crypto/libbitcoin_crypto.a /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libgmpxx.a /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libgmp.a -lboost_system -lcrypto /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/lib/libsodium.a -lrustzcash -lanl -pthread -fopenmp
/usr/bin/ld: /home/zcbbworker/ubuntu-0/ubuntu16-04/build/depends/x86_64-unknown-linux-gnu/share/../lib/librustzcash.a(rustzcash-9336abcddc89d1be.rustzcash3.rcgu.o): undefined reference to symbol 'dladdr@@GLIBC_2.2.5'
/usr/lib/gcc/x86_64-linux-gnu/5/../../../x86_64-linux-gnu/libdl.so: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

I expect the fix is the same as what I do in #2851:

```diff
diff --git a/configure.ac b/configure.ac
index f9c0f12469..27d11f766e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -695,7 +695,7 @@ AC_CHECK_LIB([gmpxx],[main],GMPXX_LIBS=-lgmpxx, [AC_MSG_ERROR(libgmpxx missing)]
 
 RUST_LIBS=""
 if test x$enable_rust != xno; then
-  RUST_LIBS="-lrustzcash"
+  RUST_LIBS="-lrustzcash -ldl"
 fi
 
 dnl Check for OpenMP support
```

File: src/gtest/main.cpp [hunk 1]
```diff
@@ -24,6 +26,8 @@ int main(int argc, char **argv) {
   boost::filesystem::path pk_path = ZC_GetParamsDir() / "sprout-proving.key";
   boost::filesystem::path vk_path = ZC_GetParamsDir() / "sprout-verifying.key";
   params = ZCJoinSplit::Prepared(vk_path.string(), pk_path.string());
+
+  TestSaplingParams = librustzcash_init_params();
```
Comment (str4d) [inline src/gtest/main.cpp]: This confused me for a moment. Could we rename it to `pTestSaplingParams`, or some other name that doesn't look like a class?

File: src/init.cpp [hunk 2]
```diff
@@ -71,6 +71,7 @@ using namespace std;
 extern void ThreadSendAlert();
 
 ZCJoinSplit* pzcashParams = NULL;
+librustzcash_params* SaplingParams = NULL;
```
Comment (str4d) [inline src/init.cpp]: Ditto (`pSaplingParams`)

File: src/test/test_bitcoin.cpp [hunk 3]
```diff
@@ -20,13 +20,16 @@
...
+#include "librustzcash.h"
+
 #include <boost/filesystem.hpp>
 #include <boost/test/unit_test.hpp>
 #include <boost/thread.hpp>
 
 CClientUIInterface uiInterface; // Declared but not defined in ui_interface.h
 CWallet* pwalletMain;
 ZCJoinSplit *pzcashParams;
+librustzcash_params* SaplingParams;
```
Comment (str4d) [inline src/test/test_bitcoin.cpp]: `pSaplingParams`

File: configure.ac [hunk 4]
```diff
@@ -728,7 +728,7 @@ AC_CHECK_LIB([gmp],[[__gmpn_sub_n]],GMP_LIBS=-lgmp, [AC_MSG_ERROR(libgmp missing
 AC_CHECK_HEADER([gmpxx.h],,AC_MSG_ERROR(libgmpxx headers missing))
 AC_CHECK_LIB([gmpxx],[main],GMPXX_LIBS=-lgmpxx, [AC_MSG_ERROR(libgmpxx missing)])
 
-RUST_LIBS="-lrustzcash"
+RUST_LIBS="-lrustzcash -ldl"
```
Comment (bitcartel) [inline configure.ac]: I guess we need this if there are undefined symbols... but I don't see "-ldl" elsewhere in the 1.1.0 source yet zcashd is already statically linked with libdl:
```
 ldd bin/zcashd
	linux-vdso.so.1 =>  (0x00007fffd618f000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fcf33f81000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fcf33bff000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fcf339fb000)
	libanl.so.1 => /lib/x86_64-linux-gnu/libanl.so.1 (0x00007fcf337f7000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fcf335ef000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fcf332e6000)
	libgomp.so.1 => /usr/lib/x86_64-linux-gnu/libgomp.so.1 (0x00007fcf330c4000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fcf32eae000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fcf32ae4000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fcf34dc4000)
```

Comment (str4d) [reply configure.ac]: `-ldl` is in the linker command [I posted above from CI](https://github.com/zcash/zcash/pull/3169#issuecomment-380984858).

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**